### PR TITLE
Level3staticfinal

### PR DESCRIPTION
**Pull request by DmitryGhoncharko**

### CODE REVIEW

**Review by AntonNovosad — APPROVED** [review]

File: src/main/java/task1/TechnicalSpecialist.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        Pattern pattern = Pattern.compile("^[0-9]{4,4} [A-Z]{2,2}-[0-9]{1,1}$");
+        Matcher matcher = pattern.matcher(number);
+        return matcher.find();
+    }
+
+    public static boolean validateModelName(String name) {
+        return name != null;
+    }
+
```
Comment (AlexeyKatomachin) [inline src/main/java/task1/TechnicalSpecialist.java:41]: remove empty row

File: src/main/java/task1/Vehicle.java [hunk 2]
```diff
@@ -23,6 +32,7 @@ public Vehicle(VehicleType vehicleType, String modelName, String registrationNum
         this.mileageKM = mileageKM;
         this.color = color;
         this.tankCapacity = tankCapacity;
+
```
Comment (AlexeyKatomachin) [inline src/main/java/task1/Vehicle.java:35]: remove empty row

File: src/main/java/task1/Vehicle.java [hunk 3]
```diff
@@ -15,6 +15,15 @@ public Vehicle() {
     }
 
     public Vehicle(VehicleType vehicleType, String modelName, String registrationNumber, double weightKG, int manufactureYear, int mileageKM, Color color, double tankCapacity) {
+        if (!(TechnicalSpecialist.validateVehicle(vehicleType) &&
```
Comment (AlexeyKatomachin) [inline src/main/java/task1/Vehicle.java:18]: separate this validation to the common method

File: src/main/java/task1/TechnicalSpecialist.java (hunk 1 above)
Comment (DmitryGhoncharko) [reply src/main/java/task1/TechnicalSpecialist.java:41]: fixed

File: src/main/java/task1/Vehicle.java (hunk 2 above)
Comment (DmitryGhoncharko) [reply src/main/java/task1/Vehicle.java:35]: fixed

File: src/main/java/task1/Vehicle.java (hunk 3 above)
Comment (DmitryGhoncharko) [reply src/main/java/task1/Vehicle.java:18]: fixed

**Review by AntonNovosad — APPROVED** [review]

**Review by AlexeyKatomachin — APPROVED** [review]

**Pull request merged**